hail/python/test/hail/matrixtable/test_file_formats.py

```diff
@@ -45,35 +47,26 @@ def all_values_table_fixture(init_hail):
     return create_all_values_table()
 
 
-# pytest sometimes uses background threads, named "Dummy-1", to collect tests. Our synchronous
-# interfaces will try to get an event loop by calling `asyncio.get_event_loop()`. asyncio will
-# create an event loop when `get_event_loop()` is called if and only if the current thread is the
-# main thread. We therefore manually create an event loop which is used only for collecting the
-# files.
-try:
-    old_loop = asyncio.get_running_loop()
-except RuntimeError as err:
-    assert 'no running event loop' in err.args[0]
-    old_loop = None
-loop = asyncio.new_event_loop()
-try:
-    asyncio.set_event_loop(loop)
-    resource_dir = resource('backward_compatability')
-    fs = hl.current_backend().fs
-    try:
-        ht_paths = [x.path for x in fs.ls(resource_dir + '/*/table/')]
-        mt_paths = [x.path for x in fs.ls(resource_dir + '/*/matrix_table/')]
-    finally:
-        hl.stop()
-finally:
-    loop.stop()
-    loop.close()
-    asyncio.set_event_loop(old_loop)
+resource_dir = resource('backward_compatability')
+def add_paths(dirname):
+    file_paths: List[str] = []
+    with os.scandir(resource_dir) as versions:
+        for version_dir in versions:
+            try:
+                with os.scandir(Path(resource_dir, version_dir, dirname)) as old_files:
+                    for file in old_files:
+                        file_paths.append(file.path)
+            except FileNotFoundError:
+                pass
+    return file_paths
+
+ht_paths = add_paths('table')
+mt_paths = add_paths('matrix_table')
```

Comment:
Using async here was quite problematic, I *think* because `pytest` uses non-main threads to run some imports so this was creating an event loop outside of the main thread. Since we don't run these tests in QoB, I just used normal synchronous python FS primitives.

Reply:
I agree with your diagnosis. Solution seems fine to me.